### ISSUE TITLE
Invalid SIP or tel URI

### ISSUE THREAD

**Issue by myfirstnameispaul** [juha-h/baresip-studio#382]
When using the default Contacts app that comes with LineageOS, I have the following in the SIP field for a contact:

    8885551212@subdomain.example.com;user=phone

When tapping, baresip reports the following error:

> Notice
> Invalid SIP or tel URI
> 'sip:8005551212%40subdomain.example.com%3Buser%3Dphone'

Version 56.3.0

**Comment by juha-h** [comment]
The report comes because the URI is invalid. @ sign in `userinfo` cannot be escaped.  Now your URI only consist of invalid `hostpart`:
```
SIP-URI          =  "sip:" [ userinfo ] hostport
                    uri-parameters [ headers ]
SIPS-URI         =  "sips:" [ userinfo ] hostport
                    uri-parameters [ headers ]
userinfo         =  ( user / telephone-subscriber ) [ ":" password ] "@"
user             =  1*( unreserved / escaped / user-unreserved )
user-unreserved  =  "&" / "=" / "+" / "$" / "," / ";" / "?" / "/"
password         =  *( unreserved / escaped /
                    "&" / "=" / "+" / "$" / "," )
hostport         =  host [ ":" port ]
host             =  hostname / IPv4address / IPv6reference
```

**Comment by juha-h** [comment]
Commit https://github.com/juha-h/baresip-studio/commit/3b3dd30875817d58861a7e3d7627ee361a603f33 should do it,but it is really ugly hack, since SIP URI userpart can contain, e.g., @ character as long as it is escaped.

**Comment by juha-h** [comment]
The hack is now included in baresip 57.1.0.
